### PR TITLE
Address more deprecation warnings

### PR DESCRIPTION
**Pull request by Eitot**
These are mostly relevant for future minimum deployment targets above macOS 10.11. However, there are some changes that affect the look of the sidebar unread count. I have not noticed any other regressions.

NSMenuItemValidation and NSToolbarItemValidation are now needed, since NSObject does not implement these and does not seem to have the corresponding validation methods anymore (at least not on macOS 11). This has exposed a bug in ButtonToolBarItem’s `validate` method that resulted in a runtime exception depending on the whether that responder implemented `validateToolbarItem` (e.g. ArticleView can be a responder and does not implement them).

### CODE REVIEW

**Comment by josh64x2** [comment]
@Eitot is it possible that this might address #1304 ?

**Review by barijaona — COMMENTED** [review]
in Light Mode, the display of unread articles count (inside folders tree) is too agressive to my eyes.

Capture d’écran 2020-11-21 à 11 22 39

**Comment by barijaona** [comment]
In fact, the problem is not due to FoldersTree, but to ImageAndTextCell's `countBackgroundColour`

**Comment by Eitot** [comment]
Yes, it seems that both the previous colour as well as `NSBackgroundStyleEmphasized` are used when a button is pressed or selected. This is why it looks light blue in bright mode. We have to chose a more appropriate colour that works well for both.

**Comment by Eitot** [comment]
You are right, it seems that the colour difference only _appears_ due the change in [FoldersTree](https://github.com/ViennaRSS/vienna-rss/pull/1372/files#diff-ad18e178b36b97071376b1d23472d89fd146e93ae3a68cf3d2b7541edcf24ca2R1018-R1031), because NSColor’s `‑colorForControlTint:` does not work at all on systems 10.14 and above. The [chosen colour in ImageAndTextCell](https://github.com/Eitot/Vienna/blob/4e3bc9b74187eec22c31813866b2346686cc88bd/Vienna/Sources/Shared/ImageAndTextCell.m#L46) (`ShadowColor`) was not shown at all on those systems. It was a bug.

The problem is more complicated, because ImageAndTextCell has its own drawing logic for the unread count. Apple usually recommends using NSButton with ‘inline style’ for this purpose.

**Comment by barijaona** [comment]
replacing shadowColor with systemGrayColor seems a safe choice

Under El Capitan:
Capture d’écran 2020-11-22 à 08 10 54

**Comment by Eitot** [comment]
It’s good enough. I had a look at Josh’s code for the NSView-based sidebar yesterday and rebased it onto master. I think the only thing that has to be implemented is the textfield for the renaming, the rest seems to work just fine.

**Review by barijaona — APPROVED** [review]
Aside the suggestions I made regarding FoldersTree background color, I am fine with this.

File: Vienna/Sources/Main window/FoldersTree.m [hunk 1]
```diff
@@ -1015,12 +1013,20 @@ -(void)outlineView:(NSOutlineView *)olv willDisplayCell:(NSCell *)cell forTableC
 		else if (folder.unreadCount)
 		{
 			[realCell setCount:folder.unreadCount];
-			[realCell setCountBackgroundColour:[NSColor colorForControlTint:[NSColor currentControlTint]]];
+            if (@available(macOS 10.14, *)) {
+                realCell.backgroundColor = NSColor.controlAccentColor;
+            } else {
+                realCell.backgroundColor = [NSColor colorForControlTint:NSColor.currentControlTint];
+            }
```
[the remark this answers is not in the document]
Comment (Eitot) [reply Vienna/Sources/Main window/FoldersTree.m:1020]: I am not actually _seeing_ realCell.backgroundColor anywhere in the interface. What is it supposed to control? I have substituted both with red and green and I am not seeing that.

[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Main window/FoldersTree.m:1020]: Good catch ! it's a NSTextFieldCell property, but obviously, many of its effects are obscured by the tests on `drawsBackground` property in ImageAndTextCell

[the remark this answers is not in the document]
Comment (Eitot) [reply Vienna/Sources/Main window/FoldersTree.m:1020]: I guess it is better then to remove those lines altogether to get rid of the warning.  The way forward would be to use NSView-based table cells as mentioned above. Maybe I can look into that soon.

**Comment by Eitot** [comment]
I added an xcconfig file to resolve the build warnings about the missing CS-ID.xcconfig file. I am using an optional include to Scripts/Resources/CS-ID.xcconfig, which means that it will include that file, if present, but not complain if it is not found. The functionality should be unaffected.

File: Vienna/Sources/Main window/FoldersTree.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Main window/FoldersTree.m]: Moving to NSView-based cells makes sense if we want to significantly overhaul the look of the table.
But I am not sure we can get similar levels of performance…

[the remark this answers is not in the document]
Comment (Eitot) [reply Vienna/Sources/Main window/FoldersTree.m]: Yes, it is unfortunate, though I think the performance is not that bad. There are plenty of Apple apps that perform well enough, at least for me. However, Apple has deprecated the NSCell-based table views. There is no other way forward, other than maybe NSCollectionView instead NSTableView.

**Review by barijaona — APPROVED** [review]

File: Vienna/Sources/Main window/FoldersTree.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Main window/FoldersTree.m]: As far as I know, NSCell-based table views are not deprecated. Some of its methods have been, but [the bulk of it is still valid](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TableView/PopulatingCellTables/PopulatingCellTables.html). Apple documentation just says : [Most Tables Are Based on NSView](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TableView/TableViewOverview/TableViewOverview.html#//apple_ref/doc/uid/10000026i-CH2-SW3)

**Pull request merged**